frontend/integration-tests/tests/kubevirt/template.wizard.scenario.ts

```diff
@@ -0,0 +1,163 @@
+/* eslint-disable no-undef, max-nested-callbacks */
+
+import { execSync } from 'child_process';
+import { browser, by, ExpectedConditions as until } from 'protractor';
+import { OrderedMap } from 'immutable';
+
+import { appHost, testName } from '../../protractor.conf';
+import { resourceRowsPresent, filterForName, deleteRow, createItemButton, isLoaded, errorMessage } from '../../views/crud.view';
+import { removeLeakedResources } from './utils';
+import { testNAD } from './mocks';
+import * as vmView from '../../views/kubevirt/vm.view';
+
+const VM_BOOTUP_TIMEOUT = 90000;
+const PAGE_LOAD_TIMEOUT = 5000;
+const VM_WIZARD_LOAD_TIMEOUT = 10000;
+
+describe('Kubevirt create VM template using wizard', () => {
+  const leakedResources = new Set<string>();
+  const vmName = `vm-${testName}`;
+  const tmplName = `template-${testName}`;
+  const operatingSystem = 'Red Hat Enterprise Linux 7.6';
+  const flavor = 'small';
+  const workloadProfile = 'generic';
+  const sourceURL = 'https://download.cirros-cloud.net/0.4.0/cirros-0.4.0-x86_64-disk.img';
+  const sourceContainer = 'kubevirt/cirros-registry-disk-demo:latest';
+  const pxeInterface = 'nic1';
+  const provisionMethods = OrderedMap<string, (provisionSource: string) => void>()
+    .set('PXE', async function(provisionSource) {
+      await vmView.provisionSourceButton.click();
+      await vmView.provisionSourceMenu.element(by.linkText(provisionSource)).click();
+    })
+    .set('Container', async function(provisionSource){
+      await vmView.provisionSourceButton.click();
+      await vmView.provisionSourceMenu.element(by.linkText(provisionSource)).click();
+
+      await vmView.provisionSourceContainerImage.sendKeys(sourceContainer);
+    })
+    .set('URL', async function(provisionSource){
+      await vmView.provisionSourceButton.click();
+      await vmView.provisionSourceMenu.element(by.linkText(provisionSource)).click();
+
+      await vmView.provisionSourceURL.sendKeys(sourceURL);
+    });
+
+  async function fillBasicSettings(provisionMethod: (provisionSource: string) => void, provisionSourceName: string, template: boolean){
+    await browser.wait(until.presenceOf(vmView.nameInput), VM_WIZARD_LOAD_TIMEOUT);
+    if (template) {
```

Comment:
Why test for template when this scenario focuses on templates? IIUC you are creating template, not a VM from it, so there is no need to have non-template related stuff. Creating VM from template may be better to include in vm.wizard.scenario.ts. ATM however requires extending Wizard class.